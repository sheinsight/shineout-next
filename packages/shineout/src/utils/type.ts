export type RequireAWithB<T, A extends keyof T, B extends keyof T> = 
  | (Omit<T, A> & { [K in A]?: never })
  | (T & Required<Pick<T, B>>)
  | (T & Required<Pick<T, A | B>>)