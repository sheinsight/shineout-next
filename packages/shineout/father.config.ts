import { defineConfig } from 'father';
export default defineConfig({
  extends: '../../.fatherrc.ts',
});
