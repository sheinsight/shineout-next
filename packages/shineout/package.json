{
  "name": "shineout",
  "version": "3.0.0",
  "description": "A components library for React",
  "keywords": [],
  "license": "ISC",
  "author": "",
  "sideEffects": false,
  "main": "./dist/cjs/index.js",
  "module": "./dist/esm/index.js",
  "typings": "./dist/cjs/index.d.ts",
  "scripts": {
    "build": "father build",
    "build:watch": "father dev",
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "dependencies": {
    "@sheinx/base": "workspace:*",
    "@sheinx/hooks": "workspace:*",
    "@sheinx/shineout-style": "workspace:*"
  },
  "peerDependencies": {
    "classnames": ">=2",
    "immer": ">=10",
    "react": ">=16.8"
  },
  "devDependencies": {
    "jszip": "^3.10.1"
  }
}
