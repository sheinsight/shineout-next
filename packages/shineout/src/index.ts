export { default as Alert } from './alert';
export { default as Button } from './button';
export { default as Checkbox } from './checkbox';
export { default as DatePicker } from './date-picker';
export { default as Dropdown } from './dropdown';
export { default as EditableArea } from './editable-area';
export { default as Form } from './form';
export { default as Icon } from './icon';
export { default as Image } from './image';
export { default as Input } from './input';
export { default as Popover } from './popover';
export { default as Radio } from './radio';
export { default as Rule } from './rule';
export { default as Spin } from './spin';
export { default as Switch } from './switch';
export { default as Tag } from './tag';
export { default as Tabs } from './tabs';
export { default as Textarea } from './textarea';
export { default as Tooltip } from './tooltip';
export { setConfig } from '@sheinx/base';

export * as TYPE from './type';
