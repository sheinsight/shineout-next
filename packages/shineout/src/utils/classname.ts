import classnames from 'classnames';
import { config } from '@sheinx/base';

interface CSSModule {
  [key: string]: string;
}

/**
 * create a new className generate function, add namespace, handle css module
 * @param style - object; for css module
 * @param module - string
 * @param prefix - string, default value is 'shineout'
 * * */
export default (style: CSSModule, module: string, prefix: string = config.prefix) =>
  (...args: any) => {
    const className = classnames(...args);
    if (!className) return '';

    const ns = `${prefix}${module ? `-${module}` : '-'}`;
    let list: (string | number)[] = className
      .split(' ')
      .map((c) => (c === '_' ? ns : `${ns}-${c}`));
    if ((config as any).cssModule) {
      list = list.map((c: any) => style[c] || c);
    }
    return list.join(' ');
  };
