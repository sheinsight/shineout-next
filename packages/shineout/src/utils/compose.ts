export function compose(...funcs: any) {
  if (funcs.length === 0) {
    return (arg: any) => arg;
  }
  const last = funcs[funcs.length - 1];
  const rest = funcs.slice(0, -1);
  return (...args: typeof last) =>
    rest.reduceRight((composed: any, f: any) => f(composed), last(...args));
}
