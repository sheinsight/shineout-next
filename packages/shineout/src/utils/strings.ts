import { util } from "@sheinx/hooks";

const { devUseWarning } = util;

export function capitalize(str: string) {
  if (typeof str !== 'string') {
    devUseWarning.error('str should be a string');
  }
  return str && str[0].toUpperCase() + str.slice(1);
}

export function substitute(str: string | Function, obj: { [x: string]: any }) {
  if (typeof str === 'string') {
    if (str.indexOf('{') < 0) {
      return str;
    }

    return str.replace(/\\?\{([^{}]+)\}/g, (match, name) => {
      if (match.charAt(0) === '\\') {
        return match.slice(1);
      }
      return obj[name] === null || obj[name] === undefined ? '' : obj[name];
    });
  }
  if (typeof str === 'function') {
    let val = str(obj);
    if (val === obj && typeof val === 'object') {
      val = Object.assign({}, obj);
    }
    return val;
  }

  return '';
}

export function removeProtocol(url: string) {
  if (url.indexOf('http') !== 0) return url;
  try {
    const { href, protocol } = new URL(url);
    return href.slice(protocol.length);
  } catch (error) {
    return url;
  }
}

export function getRTLPosition(position: string) {
  if (!position) return position;
  // position.replace('left', 'right').replace('right', 'left')
  if (position.indexOf('left') !== -1) {
    return position.replace('left', 'right');
  }
  if (position.indexOf('right') !== -1) {
    return position.replace('right', 'left');
  }
  return position;
}

export function getDirectionIconName(mode = 'left', double = false) {
  const rtl = true;
  if (mode === 'left') {
    if (rtl) {
      return double ? 'ArrowRightDouble' : 'AngleRight';
    }
    return double ? 'ArrowLeftDouble' : 'AngleLeft';
  }

  if (rtl) {
    return double ? 'ArrowLeftDouble' : 'AngleLeft';
  }
  return double ? 'ArrowRightDouble' : 'AngleRight';
}
