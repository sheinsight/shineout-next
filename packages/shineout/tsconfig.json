{
  "extends": "../../tsconfig.json",
  "compilerOptions": {
    "declaration": true,
    "baseUrl": "./",
    "paths": {
      "shineout": ["./src/index"],
      "@sheinx/base": ["../base/src/index"],
      "@sheinx/hooks": ["../hooks/src/index"]
    }
  },
  "include": ["src"]
}
