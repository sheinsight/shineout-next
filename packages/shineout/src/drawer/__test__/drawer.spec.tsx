import { cleanup, render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Drawer } from 'shineout';
import mountTest from '../../tests/mountTest';
import { classTest, createClassName, displayTest, snapshotTestByClick } from '../../tests/utils';
import DrawerBase from '../__example__/01-base';
import DrawerPosition from '../__example__/02-position';
import DrawerContainer from '../__example__/03-container';
import DrawerForm from '../__example__/04-form';
import DrawerClose from '../__example__/05-close';
import DrawerFullScreen from '../__example__/06-full-screen';
import DrawerType from '../__example__/07-type';
import DrawerHideMask from '../__example__/08-hide-mask';
import DrawerResizable from '../__example__/09-resizable';

const { wrapper, wrapperDrawerRight, wrapperDrawerTop } = createClassName(
  'modal',
  ['wrapper'],
  ['wrapperDrawerRight', 'wrapperDrawerTop'],
);

afterEach(cleanup);
mountTest(<Drawer visible />);

describe('Drawer[Base]', () => {
  displayTest(Drawer, 'ShineoutDrawer');
  snapshotTestByClick(<DrawerBase />, 'about base', wrapper);
  snapshotTestByClick(<DrawerPosition />, 'about position', wrapper);
  snapshotTestByClick(<DrawerContainer />, 'about container', wrapper);
  snapshotTestByClick(<DrawerForm />, 'about form', wrapper);
  snapshotTestByClick(<DrawerClose />, 'about close', wrapper);
  snapshotTestByClick(<DrawerFullScreen />, 'about full screen', wrapper);
  snapshotTestByClick(<DrawerType />, 'about type', wrapper);
  snapshotTestByClick(<DrawerHideMask />, 'about hide mask', wrapper);
  snapshotTestByClick(<DrawerResizable />, 'about resizable', wrapper);
  test('should render default', () => {
    render(<Drawer visible />);
    const drawerWrapper = document.querySelector(wrapper)!;
    classTest(drawerWrapper, wrapperDrawerRight);
  });
  test('should render when set position is top', () => {
    render(<Drawer visible position='top' />);
    const drawerWrapper = document.querySelector(wrapper)!;
    classTest(drawerWrapper, wrapperDrawerTop);
  });

  // all tests can be reflected in Modal components
});
