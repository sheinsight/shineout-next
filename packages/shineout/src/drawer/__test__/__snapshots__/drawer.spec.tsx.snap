// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drawer[Base] should render correctly about base 1`] = `
<div
  class="so-modal-wrapper so-modal-wrapper-animation so-modal-wrapper-show so-modal-wrapper-is-mask so-modal-wrapper-drawer so-modal-wrapper-drawer-right"
>
  <div
    class="so-modal-mask"
  >
    <div
      class="so-modal-panel"
      style="width: 400px;"
    >
      <div
        class="so-modal-header"
      >
        <div
          class="so-modal-header-title"
        >
          Drawer Title
        </div>
        <div
          class="so-modal-header-close"
        >
          <svg
            aria-hidden="true"
            fill="currentColor"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M4.6129 3.2097L4.70711 3.29289L12 10.585L19.2929 3.29289C19.6834 2.90237 20.3166 2.90237 20.7071 3.29289C21.0676 3.65338 21.0953 4.22061 20.7903 4.6129L20.7071 4.70711L13.415 12L20.7071 19.2929C21.0976 19.6834 21.0976 20.3166 20.7071 20.7071C20.3466 21.0676 19.7794 21.0953 19.3871 20.7903L19.2929 20.7071L12 13.415L4.70711 20.7071C4.31658 21.0976 3.68342 21.0976 3.29289 20.7071C2.93241 20.3466 2.90468 19.7794 3.2097 19.3871L3.29289 19.2929L10.585 12L3.29289 4.70711C2.90237 4.31658 2.90237 3.68342 3.29289 3.29289C3.65338 2.93241 4.22061 2.90468 4.6129 3.2097Z"
            />
          </svg>
        </div>
      </div>
      <div
        class="so-modal-body"
      >
        you are visited 1
      </div>
      <div
        class="so-modal-footer"
      >
        <button
          class="so-button-secondary so-button-button so-button-outline"
          type="button"
        >
          <span>
            Cancel
          </span>
        </button>
        <button
          class="so-button-primary so-button-button"
          type="button"
        >
          <span>
            Ok
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Drawer[Base] should render correctly about close 1`] = `
<div
  class="so-modal-wrapper so-modal-wrapper-animation so-modal-wrapper-show so-modal-wrapper-is-mask so-modal-wrapper-drawer so-modal-wrapper-drawer-right"
>
  <div
    class="so-modal-mask"
  >
    <div
      class="so-modal-panel"
      style="width: 400px;"
    >
      <div
        class="so-modal-header"
      >
        <div
          class="so-modal-header-title"
        >
          Modal Title
        </div>
      </div>
      <div
        class="so-modal-body"
      >
        The prop maskCloseAble is Â 
        false
        .
        <br />
        You must click the button to close the Modal.
      </div>
      <div
        class="so-modal-footer"
      >
        <button
          class="so-button-secondary so-button-button so-button-outline"
          type="button"
        >
          <span>
            Close
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Drawer[Base] should render correctly about container 1`] = `
<div
  class="so-modal-wrapper so-modal-wrapper-animation so-modal-wrapper-show so-modal-wrapper-is-mask so-modal-wrapper-drawer so-modal-wrapper-drawer-right"
>
  <div
    class="so-modal-mask"
  >
    <div
      class="so-modal-panel"
      style="width: 400px;"
    >
      <div
        class="so-modal-header"
      >
        <div
          class="so-modal-header-title"
        >
          Drawer Title
        </div>
        <div
          class="so-modal-header-close"
        >
          <svg
            aria-hidden="true"
            fill="currentColor"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M4.6129 3.2097L4.70711 3.29289L12 10.585L19.2929 3.29289C19.6834 2.90237 20.3166 2.90237 20.7071 3.29289C21.0676 3.65338 21.0953 4.22061 20.7903 4.6129L20.7071 4.70711L13.415 12L20.7071 19.2929C21.0976 19.6834 21.0976 20.3166 20.7071 20.7071C20.3466 21.0676 19.7794 21.0953 19.3871 20.7903L19.2929 20.7071L12 13.415L4.70711 20.7071C4.31658 21.0976 3.68342 21.0976 3.29289 20.7071C2.93241 20.3466 2.90468 19.7794 3.2097 19.3871L3.29289 19.2929L10.585 12L3.29289 4.70711C2.90237 4.31658 2.90237 3.68342 3.29289 3.29289C3.65338 2.93241 4.22061 2.90468 4.6129 3.2097Z"
            />
          </svg>
        </div>
      </div>
      <div
        class="so-modal-body"
      >
        Drawer mount after Button
      </div>
      <div
        class="so-modal-footer"
      >
        <button
          class="so-button-secondary so-button-button so-button-outline"
          type="button"
        >
          <span>
            Cancel
          </span>
        </button>
        <button
          class="so-button-primary so-button-button"
          type="button"
        >
          <span>
            Ok
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Drawer[Base] should render correctly about form 1`] = `
<div
  class="so-modal-wrapper so-modal-wrapper-animation so-modal-wrapper-show so-modal-wrapper-is-mask so-modal-wrapper-drawer so-modal-wrapper-drawer-right"
>
  <div
    class="so-modal-mask"
  >
    <div
      class="so-modal-panel"
      style="width: 456px;"
    >
      <div
        class="so-modal-header"
      >
        <div
          class="so-modal-header-title"
        >
          Form
        </div>
        <div
          class="so-modal-header-close"
        >
          <svg
            aria-hidden="true"
            fill="currentColor"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M4.6129 3.2097L4.70711 3.29289L12 10.585L19.2929 3.29289C19.6834 2.90237 20.3166 2.90237 20.7071 3.29289C21.0676 3.65338 21.0953 4.22061 20.7903 4.6129L20.7071 4.70711L13.415 12L20.7071 19.2929C21.0976 19.6834 21.0976 20.3166 20.7071 20.7071C20.3466 21.0676 19.7794 21.0953 19.3871 20.7903L19.2929 20.7071L12 13.415L4.70711 20.7071C4.31658 21.0976 3.68342 21.0976 3.29289 20.7071C2.93241 20.3466 2.90468 19.7794 3.2097 19.3871L3.29289 19.2929L10.585 12L3.29289 4.70711C2.90237 4.31658 2.90237 3.68342 3.29289 3.29289C3.65338 2.93241 4.22061 2.90468 4.6129 3.2097Z"
            />
          </svg>
        </div>
      </div>
      <div
        class="so-modal-body"
      >
        <form
          class="so-form-wrapper"
          style="max-width: 400px;"
        >
          <div
            class="so-form-item-wrapper so-form-item-wrapper-required"
          >
            <div
              class="so-form-item-label"
              style="width: 100px;"
            >
              Email
            </div>
            <div
              class="so-form-item-control"
            >
              <div
                class="so-input-wrapper"
                data-soui-type="input"
              >
                <div
                  class="so-input-wrapper-inner-title-top so-input-wrapper-inner-title-bottom so-input-wrapper-padding-box so-input-content"
                >
                  <input
                    class="so-input-input"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="so-form-item-wrapper so-form-item-wrapper-required"
          >
            <div
              class="so-form-item-label"
              style="width: 100px;"
            >
              Password
            </div>
            <div
              class="so-form-item-control"
            >
              <div
                class="so-input-wrapper"
                data-soui-type="input"
              >
                <div
                  class="so-input-wrapper-inner-title-top so-input-wrapper-inner-title-bottom so-input-wrapper-padding-box so-input-content"
                >
                  <input
                    class="so-input-input"
                    type="password"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
        </form>
      </div>
      <div
        class="so-modal-footer"
      >
        <div>
          <button
            class="so-button-secondary so-button-button so-button-outline"
            type="button"
          >
            <span>
              Cancel
            </span>
          </button>
          <button
            class="so-button-primary so-button-button"
            type="button"
          >
            <span>
              Submit
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Drawer[Base] should render correctly about full screen 1`] = `
<div
  class="so-modal-wrapper so-modal-wrapper-animation so-modal-wrapper-show so-modal-wrapper-is-mask so-modal-wrapper-full-screen so-modal-wrapper-drawer so-modal-wrapper-drawer-right"
>
  <div
    class="so-modal-mask"
  >
    <div
      class="so-modal-panel"
    >
      <div
        class="so-modal-header"
      >
        <div
          class="so-modal-header-title"
        >
          Profile
        </div>
        <div
          class="so-modal-header-close"
        >
          <svg
            aria-hidden="true"
            fill="currentColor"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M4.6129 3.2097L4.70711 3.29289L12 10.585L19.2929 3.29289C19.6834 2.90237 20.3166 2.90237 20.7071 3.29289C21.0676 3.65338 21.0953 4.22061 20.7903 4.6129L20.7071 4.70711L13.415 12L20.7071 19.2929C21.0976 19.6834 21.0976 20.3166 20.7071 20.7071C20.3466 21.0676 19.7794 21.0953 19.3871 20.7903L19.2929 20.7071L12 13.415L4.70711 20.7071C4.31658 21.0976 3.68342 21.0976 3.29289 20.7071C2.93241 20.3466 2.90468 19.7794 3.2097 19.3871L3.29289 19.2929L10.585 12L3.29289 4.70711C2.90237 4.31658 2.90237 3.68342 3.29289 3.29289C3.65338 2.93241 4.22061 2.90468 4.6129 3.2097Z"
            />
          </svg>
        </div>
      </div>
      <div
        class="so-modal-body"
      >
        <div>
          <form
            class="so-form-wrapper"
          >
            <div
              class="so-form-item-wrapper"
            >
              <div
                class="so-form-item-label"
              >
                name
              </div>
              <div
                class="so-form-item-control"
              >
                <div
                  class="so-input-wrapper"
                  data-soui-type="input"
                >
                  <div
                    class="so-input-wrapper-inner-title-top so-input-wrapper-inner-title-bottom so-input-wrapper-padding-box so-input-content"
                  >
                    <input
                      class="so-input-input"
                      placeholder="please input name"
                      value="zhangsan"
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              class="so-form-item-wrapper"
            >
              <div
                class="so-form-item-label"
              >
                email
              </div>
              <div
                class="so-form-item-control"
              >
                <div
                  class="so-input-wrapper"
                  data-soui-type="input"
                >
                  <div
                    class="so-input-wrapper-inner-title-top so-input-wrapper-inner-title-bottom so-input-wrapper-padding-box so-input-content"
                  >
                    <input
                      class="so-input-input"
                      placeholder="please input email"
                      value="zhangsan@qq.com"
                    />
                    <div
                      class="so-input-clear-wrapper"
                    >
                      <span
                        class="so-input-clear"
                      >
                        <svg
                          aria-hidden="true"
                          fill="currentColor"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M12 23C5.924 23 1 18.076 1 12C1 5.924 5.924 1 12 1C18.076 1 23 5.924 23 12C23 18.076 18.076 23 12 23Z"
                            fill="currentColor"
                          />
                          <path
                            d="M9.6129 8.2097C9.22061 7.90468 8.65338 7.93241 8.29289 8.29289C7.90237 8.68342 7.90237 9.31658 8.29289 9.70711L10.413 11.828L8.29289 13.9497L8.2097 14.044C7.90468 14.4362 7.93241 15.0035 8.29289 15.364C8.68342 15.7545 9.31658 15.7545 9.70711 15.364L11.828 13.242L13.9497 15.364L14.044 15.4471C14.4362 15.7522 15.0035 15.7244 15.364 15.364C15.7545 14.9734 15.7545 14.3403 15.364 13.9497L13.242 11.828L15.364 9.70711L15.4471 9.6129C15.7522 9.22061 15.7244 8.65338 15.364 8.29289C14.9734 7.90237 14.3403 7.90237 13.9497 8.29289L11.828 10.413L9.70711 8.29289L9.6129 8.2097Z"
                            fill="white"
                          />
                        </svg>
                      </span>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="so-form-item-wrapper"
            >
              <div
                class="so-form-item-label"
              />
              <div
                class="so-form-item-control"
              >
                <button
                  type="submit"
                >
                  æäº¤
                </button>
                <button
                  type="reset"
                >
                  éç½®
                </button>
              </div>
            </div>
          </form>
        </div>
      </div>
      <div
        class="so-modal-footer"
      >
        <button
          class="so-button-primary so-button-button"
          type="button"
        >
          <span>
            OK
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Drawer[Base] should render correctly about hide mask 1`] = `
<div
  class="so-modal-wrapper so-modal-wrapper-animation so-modal-wrapper-show so-modal-wrapper-is-mask so-modal-wrapper-hide-mask so-modal-wrapper-drawer so-modal-wrapper-drawer-right"
>
  <div
    class="so-modal-mask"
  >
    <div
      class="so-modal-panel"
      style="width: 400px;"
    >
      <div
        class="so-modal-header"
      >
        <div
          class="so-modal-header-title"
        >
          Drawer Title
        </div>
        <div
          class="so-modal-header-close"
        >
          <svg
            aria-hidden="true"
            fill="currentColor"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M4.6129 3.2097L4.70711 3.29289L12 10.585L19.2929 3.29289C19.6834 2.90237 20.3166 2.90237 20.7071 3.29289C21.0676 3.65338 21.0953 4.22061 20.7903 4.6129L20.7071 4.70711L13.415 12L20.7071 19.2929C21.0976 19.6834 21.0976 20.3166 20.7071 20.7071C20.3466 21.0676 19.7794 21.0953 19.3871 20.7903L19.2929 20.7071L12 13.415L4.70711 20.7071C4.31658 21.0976 3.68342 21.0976 3.29289 20.7071C2.93241 20.3466 2.90468 19.7794 3.2097 19.3871L3.29289 19.2929L10.585 12L3.29289 4.70711C2.90237 4.31658 2.90237 3.68342 3.29289 3.29289C3.65338 2.93241 4.22061 2.90468 4.6129 3.2097Z"
            />
          </svg>
        </div>
      </div>
      <div
        class="so-modal-body"
      >
        you are visited 1
      </div>
      <div
        class="so-modal-footer"
      >
        <button
          class="so-button-secondary so-button-button so-button-outline"
          type="button"
        >
          <span>
            Cancel
          </span>
        </button>
        <button
          class="so-button-primary so-button-button"
          type="button"
        >
          <span>
            Ok
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Drawer[Base] should render correctly about position 1`] = `
<div
  class="so-modal-wrapper so-modal-wrapper-animation so-modal-wrapper-show so-modal-wrapper-is-mask so-modal-wrapper-drawer so-modal-wrapper-drawer-right"
>
  <div
    class="so-modal-mask"
  >
    <div
      class="so-modal-panel"
      style="width: auto;"
    >
      <div
        class="so-modal-header"
      >
        <div
          class="so-modal-header-title"
        >
          Form
        </div>
        <div
          class="so-modal-header-close"
        >
          <svg
            aria-hidden="true"
            fill="currentColor"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M4.6129 3.2097L4.70711 3.29289L12 10.585L19.2929 3.29289C19.6834 2.90237 20.3166 2.90237 20.7071 3.29289C21.0676 3.65338 21.0953 4.22061 20.7903 4.6129L20.7071 4.70711L13.415 12L20.7071 19.2929C21.0976 19.6834 21.0976 20.3166 20.7071 20.7071C20.3466 21.0676 19.7794 21.0953 19.3871 20.7903L19.2929 20.7071L12 13.415L4.70711 20.7071C4.31658 21.0976 3.68342 21.0976 3.29289 20.7071C2.93241 20.3466 2.90468 19.7794 3.2097 19.3871L3.29289 19.2929L10.585 12L3.29289 4.70711C2.90237 4.31658 2.90237 3.68342 3.29289 3.29289C3.65338 2.93241 4.22061 2.90468 4.6129 3.2097Z"
            />
          </svg>
        </div>
      </div>
      <div
        class="so-modal-body"
      >
        <form
          class="so-form-wrapper"
          style="width: 500px;"
        >
          <div
            class="so-form-item-wrapper so-form-item-wrapper-required"
          >
            <div
              class="so-form-item-label"
              style="width: 100px;"
            >
              Email
            </div>
            <div
              class="so-form-item-control"
            >
              <div
                class="so-input-wrapper"
                data-soui-type="input"
              >
                <div
                  class="so-input-wrapper-inner-title-top so-input-wrapper-inner-title-bottom so-input-wrapper-padding-box so-input-content"
                >
                  <input
                    class="so-input-input"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="so-form-item-wrapper so-form-item-wrapper-required"
          >
            <div
              class="so-form-item-label"
              style="width: 100px;"
            >
              Password
            </div>
            <div
              class="so-form-item-control"
            >
              <div
                class="so-input-wrapper"
                data-soui-type="input"
              >
                <div
                  class="so-input-wrapper-inner-title-top so-input-wrapper-inner-title-bottom so-input-wrapper-padding-box so-input-content"
                >
                  <input
                    class="so-input-input"
                    type="password"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
        </form>
      </div>
      <div
        class="so-modal-footer"
      >
        <div>
          <button
            class="so-button-secondary so-button-button so-button-outline"
            type="button"
          >
            <span>
              Cancel
            </span>
          </button>
          <button
            class="so-button-primary so-button-button"
            type="button"
          >
            <span>
              Submit
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Drawer[Base] should render correctly about resizable 1`] = `
<div
  class="so-modal-wrapper so-modal-wrapper-animation so-modal-wrapper-show so-modal-wrapper-is-mask so-modal-wrapper-drawer so-modal-wrapper-drawer-right"
>
  <div
    class="so-modal-mask"
  >
    <div
      class="so-modal-panel"
      style="width: auto;"
    >
      <div
        class="so-modal-header"
      >
        <div
          class="so-modal-header-title"
        >
          Form
        </div>
        <div
          class="so-modal-header-close"
        >
          <svg
            aria-hidden="true"
            fill="currentColor"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M4.6129 3.2097L4.70711 3.29289L12 10.585L19.2929 3.29289C19.6834 2.90237 20.3166 2.90237 20.7071 3.29289C21.0676 3.65338 21.0953 4.22061 20.7903 4.6129L20.7071 4.70711L13.415 12L20.7071 19.2929C21.0976 19.6834 21.0976 20.3166 20.7071 20.7071C20.3466 21.0676 19.7794 21.0953 19.3871 20.7903L19.2929 20.7071L12 13.415L4.70711 20.7071C4.31658 21.0976 3.68342 21.0976 3.29289 20.7071C2.93241 20.3466 2.90468 19.7794 3.2097 19.3871L3.29289 19.2929L10.585 12L3.29289 4.70711C2.90237 4.31658 2.90237 3.68342 3.29289 3.29289C3.65338 2.93241 4.22061 2.90468 4.6129 3.2097Z"
            />
          </svg>
        </div>
      </div>
      <div
        class="so-modal-body"
      >
        <form
          class="so-form-wrapper"
          style="width: 500px;"
        >
          <div
            class="so-form-item-wrapper so-form-item-wrapper-required"
          >
            <div
              class="so-form-item-label"
              style="width: 100px;"
            >
              Email
            </div>
            <div
              class="so-form-item-control"
            >
              <div
                class="so-input-wrapper"
                data-soui-type="input"
              >
                <div
                  class="so-input-wrapper-inner-title-top so-input-wrapper-inner-title-bottom so-input-wrapper-padding-box so-input-content"
                >
                  <input
                    class="so-input-input"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="so-form-item-wrapper so-form-item-wrapper-required"
          >
            <div
              class="so-form-item-label"
              style="width: 100px;"
            >
              Password
            </div>
            <div
              class="so-form-item-control"
            >
              <div
                class="so-input-wrapper"
                data-soui-type="input"
              >
                <div
                  class="so-input-wrapper-inner-title-top so-input-wrapper-inner-title-bottom so-input-wrapper-padding-box so-input-content"
                >
                  <input
                    class="so-input-input"
                    type="password"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
        </form>
      </div>
      <div
        class="so-modal-footer"
      >
        <div>
          <button
            class="so-button-secondary so-button-button so-button-outline"
            type="button"
          >
            <span>
              Cancel
            </span>
          </button>
          <button
            class="so-button-primary so-button-button"
            type="button"
          >
            <span>
              Submit
            </span>
          </button>
        </div>
      </div>
      <div
        class="so-modal-resize-x"
        data-position="right"
      />
    </div>
  </div>
</div>
`;

exports[`Drawer[Base] should render correctly about type 1`] = `
<div
  class="so-modal-wrapper so-modal-wrapper-animation so-modal-wrapper-show so-modal-wrapper-is-mask so-modal-wrapper-drawer so-modal-wrapper-drawer-right"
>
  <div
    class="so-modal-mask"
  >
    <div
      class="so-modal-panel"
      style="width: 500px;"
    >
      <div
        class="so-modal-header"
      >
        <span
          class="so-modal-header-icon so-alert-icon so-alert-success-icon"
        >
          <svg
            aria-hidden="true"
            fill="currentColor"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 23C5.92487 23 1 18.0751 1 12C1 5.92487 5.92487 1 12 1C18.0751 1 23 5.92487 23 12C23 18.0751 18.0751 23 12 23ZM6.34315 13.8284L8.46305 15.9483C9.24511 16.7304 10.5122 16.7304 11.2943 15.9483L17.6569 9.58579C18.0472 9.19546 18.0472 8.5619 17.6569 8.17157C17.2665 7.78125 16.633 7.78125 16.2426 8.17157L10.5872 13.827C10.1962 14.218 9.56119 14.218 9.17016 13.827L7.75736 12.4142C7.36704 12.0239 6.73347 12.0239 6.34315 12.4142C5.95282 12.8045 5.95282 13.4381 6.34315 13.8284Z"
            />
          </svg>
        </span>
        <div
          class="so-modal-header-title"
        >
          Drawer Title with success Icon
        </div>
        <div
          class="so-modal-header-close"
        >
          <svg
            aria-hidden="true"
            fill="currentColor"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M4.6129 3.2097L4.70711 3.29289L12 10.585L19.2929 3.29289C19.6834 2.90237 20.3166 2.90237 20.7071 3.29289C21.0676 3.65338 21.0953 4.22061 20.7903 4.6129L20.7071 4.70711L13.415 12L20.7071 19.2929C21.0976 19.6834 21.0976 20.3166 20.7071 20.7071C20.3466 21.0676 19.7794 21.0953 19.3871 20.7903L19.2929 20.7071L12 13.415L4.70711 20.7071C4.31658 21.0976 3.68342 21.0976 3.29289 20.7071C2.93241 20.3466 2.90468 19.7794 3.2097 19.3871L3.29289 19.2929L10.585 12L3.29289 4.70711C2.90237 4.31658 2.90237 3.68342 3.29289 3.29289C3.65338 2.93241 4.22061 2.90468 4.6129 3.2097Z"
            />
          </svg>
        </div>
      </div>
      <div
        class="so-modal-body so-modal-body-with-icon"
      >
        <span>
          Drawer type: 
        </span>
        <b>
          success
        </b>
      </div>
      <div
        class="so-modal-footer"
      >
        <button
          class="so-button-secondary so-button-button so-button-outline"
          type="button"
        >
          <span>
            Cancel
          </span>
        </button>
        <button
          class="so-button-primary so-button-button"
          type="button"
        >
          <span>
            Ok
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
